sherpa/utils/__init__.py

```diff
@@ -306,15 +309,39 @@ def has_package_from_list(*packages):
     return False
 
 
-def has_fits_support():
+def requires_package(msg=None, *packages):
+    """
+    Decorator for test functions requiring specific packages.
+    """
+    condition = _has_package_from_list(*packages)
+    msg = msg or "required module missing among {}.".format(", ".join(packages))
+
+    def decorator(test_function):
+        return skipIf(not condition, msg)(test_function)
+    return decorator
+
+
+def requires_xspec(test_function):
+    """Decorator for test functions requiring xspec"""
+    return requires_package('xspec required', 'sherpa.astro.xspec')(test_function)
+
+
+def requires_ds9(test_function):
+    """Decorator for test functions requiring ds9"""
+    return requires_package('ds9 required', 'sherpa.image.ds9_backend')(test_function)
+
+
+def requires_fits(test_function):
     """
     Returns True if there is an importable backend for FITS I/O.
     Used to skip tests requiring fits_io
     """
-    return has_package_from_list('pyfits',
-                                 'astropy.io.fits',
-                                 'pycrates',
-                                 )
+    packages = ('pyfits',
```

Comment:
Hmm... I don't think that makes a difference, actually. In case someone has both installed, the condition will be satisfied by pyfits, but then astropy will be selected as a backend. The purpose of this check is just to understand whether a test has to be run or skipped, not which backend to load.